src/widget/form/settings/avform.cpp

```diff
@@ -116,6 +117,12 @@ void AVForm::showEvent(QShowEvent* event)
     createVideoSurface();
     getVideoDevices();
 
+    // To don't block GUI thread.
+    QtConcurrent::run([this]()
+    {
+        camVideoSurface->setSource(&camera);
```

Comment:
Straight call qwidget from another thread usually is bad practice. Are you sure that setSource is thread-safe?